gcloud-java-core/src/main/java/com/google/gcloud/BatchResult.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+/**
+ * The class that holds a single result of a batch call.
+ */
+public interface BatchResult<T, E extends Throwable> {
```

Comment:
Shouldn't `E` extend `BaseServiceException`?

Also, shouldn't this class also expose a `notify` method to register a callback?
```
void notify(Callback<T> callback);
```